data/binding/printf.go

```diff
@@ -0,0 +1,254 @@
+package binding
+
+import (
+	"fmt"
+)
+
+type sprintfString struct {
+	String
+
+	format string
+	source []interface{}
+	err    error
+}
+
+// NewSprintf returns a String binding that format its content using the
+// format string and the provide additional parameter that must be other
+// data bindings.
+//
+// Since: 2.2
+func NewSprintf(format string, b ...interface{}) (String, error) {
```

Comment:
If the additional parameter must be a data binding, why not use `DataItem` here?
```suggestion
func NewSprintf(format string, b ...DataItem) (String, error) {
```

Reply:
I used interface{} to copy the prototype of sprintf, but I agree a DataItem should work. Will update.